src/commands/login.rs

```diff
@@ -256,10 +264,29 @@ impl LoginCommand {
         let login_connection = match self.auth_method() {
             AuthMethod::Github => self.run_interactive_gh_login().await?,
             AuthMethod::UsernameAndPassword => self.run_interactive_basic_login().await?,
+            AuthMethod::Token => self.login_using_token().await?,
         };
         self.save_login_info(&login_connection)
     }
 
+    async fn login_using_token(&self) -> Result<LoginConnection> {
+        HippoClient::new(ConnectionInfo {
+            url: self.hippo_server_url.to_string(),
+            danger_accept_invalid_certs: self.insecure,
+            api_key: self.token.clone(),
+        })
+        .list_apps()
```

Comment:
Why is listing apps part of the login flow?

Reply:
We don't have a dedicated API to verify auth info.

Reply:
Oh, so we're verifying the operation succeeds. Got it. I think it's worth adding a comment explaining that.

Reply:
added the comment